functions/snippets/src/main/java/Tips.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START functions_tips_scopes]
+// [START run_tips_global_scope]
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class Tips {
+
+  // Create a client with some reasonable defaults. This client can be reused for multiple requests.
+  private static HttpClient client =
+      HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(10)).build();
+
+  // [END functions_tips_scopes]
+  // [END run_tips_global_scope]
+  private static int lightComputation() {
+      List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9);
+      return numbers.stream().reduce((t, x) -> t + x).get();
+  }
+
+  private static int heavyComputation() {
+      List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9);
+      return numbers.stream().reduce((t, x) -> t + x).get();
+  }
+
+  // [START functions_tips_scopes]
+  // [START run_tips_global_scope]
+  // Global (instance-wide) scope
+  // This computation runs at instance cold-start
+  public static final int INSTANCE_VAR = heavyComputation();
```

Comment:
Class variables should be declared at the top of the class.

Reply:
Also, this might not be the way we want to showcase this. Normally, global variables aren't used with servlets due to issues with thread-safety. This example is probably okay because it's a final variable, but we probably either want to add a warning or (better yet) update the example to us a `ServletContextListener` for creating start up resources.

Reply:
Moved to the very top. Not sure if we need the context listener.

Reply:
It should be updated to either use a ContextListener, or include a warning that global values in Servlets are not threadsafe, and a `ContextListener` is preferred for mutable variables. 

I'm not really sure if/how a ContextListener would work with functions, so this snippet might be a good example to showcase that pattern as well.

Reply:
I don't really know how to proceed. Do you think that GCF will use multiple threads per Function invocation?

> This example is probably okay because it's a final variable

Since this is a `final` variable, isn't it safe to access across multiple threads?

Reply:
Most servlet environments will use multiple threads to handle requests, so

> Do you think that GCF will use multiple threads per Function invocation?

I don't know how GCF handles things, but it would be idiomatic to Java to remain threadsafe when using Servlets since other runtimes may use multiple threads. 

> Since this is a final variable, isn't it safe to access across multiple threads?

I should have been more clear - since this is a `final` and a primitive it's immutable, and thus thread-safe. However, we should either point out _why_ this is okay, or show the user how to handle these properly. If the user tried to do this with an object that isn't thread-safe (even if it's final), they'll run into issues.